lib/configuration/variables/sources/merge.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const ServerlessError = require('../../../serverless-error');
+const isPlainObject = require('type/plain-object/is');
+
+module.exports = {
+  resolve: ({ params }) => {
+    if (!params || params.length < 2) {
+      throw new ServerlessError('${merge(...)} needs a sequence of values to merge');
+    }
+
+    if (params.every((item) => Array.isArray(item))) {
+      const result = [];
+      for (const param of params) result.push(...param);
+      return { value: result };
+    }
+
+    if (params.every((item) => isPlainObject(item))) {
+      const keys = {};
+
+      params.forEach((item, index) => {
+        Object.keys(item).forEach((key) => {
+          if (keys[key] !== undefined) {
+            throw new ServerlessError(
+              `duplicate key \`${key}\` in \${merge(...)}[${index}]; first seen in \${merge(...)}[${keys[key]}]`
```

Comment:
Let's reword to ``Unsafe arguments for variable "merge" source: Duplicate key "${key}" in [${index}] (first seen in [${keys[key]}])``, and add code `UNSAFE_VARIABLE_MERGE_PARAMS`